File: multiplication/src/main/java/learn/mssb/mssbmultiplication/challenge/ChallengeAttemptDto.java
package learn.mssb.mssbmultiplication.challenge;

import lombok.Value;

import javax.validation.constraints.Max;
import javax.validation.constraints.Min;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.Positive;

@Value
public class ChallengeAttemptDto {
    @NotBlank
    String userAlias;
    @Min(1) @Max(99)
    int factorA;
    @Min(1) @Max(99)
    int factorB;
    @Positive(message = "You can not get a negative result here. Try again")
    int guess;
}

File: multiplicationkt/src/main/kotlin/challenge/ChallengeAttemptDTO.kt
package learn.mssbkt.multiplicationkt.challenge

import jakarta.validation.constraints.Max
import jakarta.validation.constraints.Min
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.Positive

/**
 * Попытка решения пользователя.
 */
data class ChallengeAttemptDTO(
    @field:Min(1)
    @field:Max(99)
    val factorA: Int,
    @field:Min(1)
    @field:Max(99)
    val factorB: Int,
    @field:NotBlank
    val userAlias: String,
    @field:Positive(message = "How could you possibly get a negative result here? Try again.")
    val guess: Int,
)
